packages/core/lib/remote-parent-context.ts

```diff
@@ -0,0 +1,43 @@
+import { scaleProbabilityToMatchSamplingRate } from './sampler'
+import type { ParentContext } from './span'
+import { isParentContext, isSpanContext } from './validation'
+
+const TRACE_PERENT_REGEX = /^00-([0-9a-f]{32})-([0-9a-f]{16})-[0-9]{2}$/
+
+export default class RemoteParentContext implements ParentContext {
+  readonly id: string
+  readonly traceId: string
+
+  constructor (maybeParentContext: ParentContext | string, traceId?: string) {
+    if (isParentContext(maybeParentContext)) {
+      this.id = maybeParentContext.id
+      this.traceId = maybeParentContext.traceId
+    } else {
+      this.id = maybeParentContext
+      this.traceId = traceId || ''
+    }
+  }
+
+  encodeAsTraceParent () {
+    return RemoteParentContext.toTraceParentString(this)
+  }
+
+  static parseTraceParent (encodedString: string) {
+    if (!TRACE_PERENT_REGEX.test(encodedString)) {
+      return undefined
+    }
+
+    const [, traceId, spanId] = encodedString.split('-')
+    return new RemoteParentContext(spanId, traceId)
```

Comment:
nit: use the captured groups in the regex instead of splitting the string

```suggestion
    const result = TRACE_PERENT_REGEX.exec(encodedString)
    if (!result) {
      return undefined
    }

    const [, traceId, spanId] = result
    return new RemoteParentContext(spanId, traceId)
```